### PR TITLE
feature(StatusAddressPanel): Implement generic address display control

### PR DESCRIPTION
**Pull request by stefandunca**
### Showcase

Some images and video is not up to date, tweaked and fixed in the meantime

#### App

Light theme
image

Dark theme
image

StatusQ vs. Design
image

Hovered
image

Wallet settings
image

#### Sandbox
https://user-images.githubusercontent.com/47554641/182799214-505900a3-5f43-48f4-b28e-822f4e084015.mov

### Checklist

- [x] follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] add documentation if necessary (new component, new feature)
- [x] update sandbox app
- [x] test changes in both light and dark theme?
- [x] ~~is this a breaking change?~~
- [x] test changes in [status-desktop](https://github.com/status-im/status-desktop)

### CODE REVIEW

**Review by MishkaRogachev — APPROVED** [review]
Layouts are cool

File: src/StatusQ/Components/StatusAddressPanel.qml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+    MouseArea {
+        id: mainMouseArea
+
+        anchors.fill: parent
+
+        hoverEnabled: root.expandable || root.autHideCopyIcon
+        enabled: (root.autHideCopyIcon && root.showCopy) || root.expandable
+        acceptedButtons: root.expandable ? Qt.LeftButton : Qt.NoButton
+        cursorShape: root.expandable ? Qt.PointingHandCursor : Qt.arrowCursor
```
Comment (caybro) [inline src/StatusQ/Components/StatusAddressPanel.qml:184]: `Qt.ArrowCursor` typo

File: src/StatusQ/Components/StatusAddress.qml [hunk 2]
```diff
@@ -12,13 +13,14 @@ StatusBaseText {
     readonly property real actualWidth: implicitWidth
     property real maxWidth: width
 
+    property int preferredWidth: expanded ? actualWidth : maxWidth
```
Comment (caybro) [inline src/StatusQ/Components/StatusAddress.qml:16]: `readonly`?

**Review by caybro — CHANGES_REQUESTED** [review]
Looks good, some minor stuff only

**Review by caybro — APPROVED** [review]
Good stuff!

**Pull request merged**